phpBB/phpbb/install/module_base.php

```diff
@@ -110,10 +110,38 @@ public function run()
 
 		if ($task_index < $iterator->count())
 		{
-			$iterator->seek($task_index);
+		    // Receiving task index
+		    $get_task_index = (int)$this->iohandler->get_input('task_index', 0);
+		    // Validation check
+		    if($get_task_index > 0 && $get_task_index < $iterator->count()){
+                $get_task_index--;
+                $iterator->seek($get_task_index);
+                // Setting the menu point as active
+                $this->iohandler->set_active_stage_menu($iterator->current()->get_navigation_stage_path());
+                $this->iohandler->send_response();
+            }
+            else{
+                // Displaying completed as checked
+                for($i = 0; $i < $task_index; $i++){
```

Comment:
You can use a `foreach` here, it would tidy it up. You instantiate a bunch of tasks just to retrieve an array to send some info to the client side. You already have that information on the client side. The user already clicked a link in a list, making the clicked item active, and every subsequent one inactive. There is not really any reason to do this IMHO.